cit3.0-web/src/components/MapContainer/MapContainer.js

```diff
@@ -1,48 +1,63 @@
 import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
+import axios from "axios";
 import { useDispatch } from "react-redux";
 import Map from "../Map/Map";
 import { getProximityData } from "../../helpers/resourceCalls";
 import { getSoilAndElevationData, buildSoilString } from "../../helpers/soil";
-import { setSoil, setElevation } from "../../store/actions/opportunity";
-
-const resourceIds = {
-  Hospitals: "5ff82cf4-0448-4063-804a-7321f0f2b4c6",
-  Schools: "5832eff2-3380-435e-911b-5ada41c1d30b",
-  "Post Secondary Schools": "8e4e2a87-2d1d-4931-828e-6327b49f310e",
-  Courts: "23aa0b75-2715-4ccb-9a36-9a608450dc2d",
-  "Walk-In Clinics": "3ca6b086-c92b-4654-ae82-ff5723d00611",
-  "Natural Resource Projects": "2b69cc4b-4076-4272-a5a0-1c731455e063",
-  "Economic Projects": "b12cd4cc-b58b-4079-b630-a20b6df58e8d",
-};
+import {
+  setSoil,
+  setElevation,
+  setNearbyResources,
+} from "../../store/actions/opportunity";
 
 export default function MapContainer({
   nearbyResources,
-  setNearbyResources,
   coords,
   setCoords,
   setAddress,
   setError,
   setNoAddressFlag,
+  setProximityInProgress,
 }) {
   const [lastCoords, setLastCoords] = useState([]);
   const dispatch = useDispatch();
 
-  const run = async () => {
-    const soilData = await getSoilAndElevationData(coords);
+  const { CancelToken } = axios;
+  let source = CancelToken.source();
+
+  const run = async (sourceToken) => {
+    // proximity call is running, do not unmount component if user hits continue
+    setProximityInProgress(true);
+    console.log("running");
+    const [soilData, proximityData] = await axios.all([
```

Comment:
I would prefer that the axios call be in an actions file or something like that. I just like to keep all of the API calls together.